week1/Q2.py

```diff
@@ -0,0 +1,51 @@
+class Node:
+	def __init__(self, value = None, next = None):
+		self.value = value
+		self.next = next
+
+
+class LinkedList:
+	def __init__(self):
+		self.first = None
+		self.last = None
+		self.length = 0
+
+	def clear(self):
+		self.__init__()
+
+	def add(self, x):
+		self.length += 1
+		if self.first == None:
+			self.last = self.first = Node(x, None)
+		else:
+			self.last.next = self.last = Node(x, None)
```

Comment:
While this works, I think this is rather cryptic and it depends on the evaluation order of the sub-expressions.
Sometime cool looking code is actually much harder to understand or can become buggy when modified.
I would recommend splitting this into multiple statements.